From 6ae89eee936159dd7d7ca3405fbe6d9d66d86ddd Mon Sep 17 00:00:00 2001
From: Christoph Feck <christoph@maxiom.de>
Date: Mon, 7 Nov 2011 21:29:38 +0100
Subject: [PATCH] Fix crash in Folder destructor

Especially visible with Qt 4.8.0 qDeleteAll() changes, but might also
fix older bugs.

BUG: 285938
CCBUG: 193780
FIXED-IN: 4.7.4
REVIEW: 102981
(cherry picked from commit 3497e5afe51490191825c7b7d475a5fc0702988d)
---
 akregator/src/folder.cpp |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/akregator/src/folder.cpp b/akregator/src/folder.cpp
index f2385b0..249e598 100644
--- a/akregator/src/folder.cpp
+++ b/akregator/src/folder.cpp
@@ -65,7 +65,10 @@ Folder::FolderPrivate::FolderPrivate( Folder* qq ) : q( qq ), unread( 0 ), open(
 
 Folder::FolderPrivate::~FolderPrivate()
 {
-    qDeleteAll( children );
+    while (!children.isEmpty()) {
+        // child removes itself from list in its destructor
+        delete children.first();
+    }
     emit q->emitSignalDestroyed();
 }
 
-- 
1.7.10

